core/src/main/java/com/amplifyframework/util/UserAgent.java

```diff
@@ -235,7 +235,24 @@ public String toString() {
 
         @NonNull
         private static String sanitize(@Nullable String string) {
-            return string != null ? string : "UNKNOWN";
+            return string != null ? stripHeader(string) : "UNKNOWN";
+        }
+
+        /**
+         * Strips non-permitted characters in the header by okhttp.
+         */
+        private static String stripHeader(String value) {
+            StringBuilder builder = new StringBuilder(value.length());
+            boolean modified = false;
+            for (int i = 0, length = value.length(); i < length; i++) {
+                char c = value.charAt(i);
+                if ((c > '\u001f' && c < '\u007f') || c == '\t') {
```

Comment:
Could you add a comment here I have no idea what these UTF values are 😄

Reply:
it is filtering out non-ascii values from the string.